net/Socket.cpp

```diff
@@ -333,26 +334,29 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS)
     setupPollFds(now, timeoutMaxMicroS);
     const size_t size = _pollSockets.size();
 
+    bool hasURP = _urpFD != -1;
```

Comment:
I'm a tad surprised we can't use a normal StreamSocket for the URP  thing and add that to the SocketPoll in the normal way - did you try without this ? =)